pkg/proto/aksnodeconfig/v1/bootstrappingconfig.proto

```diff
@@ -0,0 +1,43 @@
+syntax = "proto3";
+package aksnodeconfig.v1;
+
+message BootstrappingConfig {
+  // Method to authenticate the node to the API server
+  BootstrappingAuthMethod bootstrapping_auth_method = 1;
+
+  // how the node should join and communicate with the API server after authentication
+  ClusterJoinMethod cluster_join_method = 2;
+
+  // Only required until Secure TLS bootstrapping in place. Would use kubelet identity after that.
+  optional string tls_bootstrapping_token = 3;
+
+  // Only used when secure TLS bootstrapping is enabled or one of the Azure/Arc methods. This is the appserver appid that the node will use to bootstrap.
+  optional string custom_aad_resource = 4;
+
+  // Only used when one of the Azure/Arc methods is enabled. This is the client ID of the MSI that the node will use to bootstrap.
+  optional string custom_aad_client_id = 5;
+
+  enum BootstrappingAuthMethod  {
```

Comment:
Add an unspecified option. So, we can distinguish between user intentionally setting BOOTSTRAP_TOKEN vs default value.